app/src/main/java/com/hfad/mymessenger/CreateMessageActivity.java

```diff
@@ -20,18 +20,33 @@ protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_create_message);
         //TODO: assign findViewById values to all instance variables
-        messageView = (EditText)findViewById(R.id.message);
+        buttonToApp = (Button) findViewById(R.id.sendExplicit);
+        buttonToActivity = (Button) findViewById(R.id.sendImplicit);
+        messageView = (EditText) findViewById(R.id.message);
+
 
-        
         //TODO: Add a setOnClickListener to each Button
+        buttonToApp.setOnClickListener(new View.OnClickListener() {
+            public void onClick(View v) {
+                sendMessageToApp();
+            }
+        });
+
+        buttonToActivity.setOnClickListener(new View.OnClickListener() {
+            public void onClick(View v) {
+                sendMessageToActivity();
+            }
+        });
+    }
 
 
-    }
 
     //Call sendMessageToActivity when the "to Activity" button is clicked
     public void sendMessageToActivity(){
         //TODO: Use an explicit Intent to invoke the ReceiveMessageActivity with the message
-
+        Intent intent = new Intent(this, ReceiveMessageActivity.class);
+        intent.putExtra("message", true);
```

Comment:
Include the message text as well, not just a boolean value. That way you can populate the text in the Implicit intent